asset outdated warning inline on full screen

brad-decker opened the pull request:
Work towards implementing the full-screen design for the asset page

This PR:
1. Updates the list-item to expose a new prop `midContent` which I can imagine might be used in other use cases as well. 
2. Updates the styling of list-item to be responsive with/without `midContent` present.
3. repurposes `warning` prop on asset-list-item to be multipurpose so that it can be used to build the tooltip and inline experience.
4. Updates one thing with transaction list item to update font-sizes on the transaction list item to match designs.

Does not include:
1. The send token button on full screen UI
2. Update to the text of the warning to indicate last sync time.

Screenshots will be added once CI passes


<details>
<summary>Asset page Responsive Gif</summary>
<img src="https://user-images.githubusercontent.com/4448075/83788891-8a416e00-a65b-11ea-922c-0f674d2c2690.gif" />
</details>

<details>
<summary>Transaction list item screenshot before</summary>
<img src="https://user-images.githubusercontent.com/4448075/83789460-5f0b4e80-a65c-11ea-9e88-b513bcac4c0a.png" />
</details>

<details>
<summary>Transaction list item screenshot after</summary>
<img src="https://user-images.githubusercontent.com/4448075/83789273-1784c280-a65c-11ea-9be7-70530ac2772d.png" />
</details>

Gudahtt commented on ui/app/components/app/asset-list-item/asset-list-item.scss:
```diff
@@ -7,4 +7,24 @@
   }
+
+  &__warning {
+    flex: 1;
```
Nit: I suspect this rule might not do anything? Since the `__warning` component only has a single child.

```suggestion
```

brad-decker replied:
flex: 1 is applied to the element itself, the parent is display flex, this makes explicit the sizing rules for the two components (the warning and the icon) where previously it was implicit and causes the icon to be resized to fit a gradually changing content box. by telling the __warning element to take up at most as much space that is not occupied by other flex items, it keeps the icon taking up it's appropriate space and allows the text to fill in the remainder. 
![warning](https://user-images.githubusercontent.com/4448075/83904353-55015280-a725-11ea-938f-3c6d3b51192c.gif)

Gudahtt replied:
Huh. I did try that, but I must not have tried the correct screen size. Thanks, that makes sense.

Gudahtt commented on ui/app/components/ui/list-item/index.scss:
```diff
@@ -11,19 +11,28 @@
 
-  &__icon > * {
-    margin: 8px 14px 0 0;
+  &__icon {
+    display: flex;
+    align-items: center;
+    > * {
+      margin: 0 16px 0 0;
+    }
```
Nit: I'm not sure these rules make a difference either

```suggestion
  &__icon > * {
    margin: 0 16px 0 0;
  }
```

brad-decker replied:
Those rules are how the icon is being centered.
![icon](https://user-images.githubusercontent.com/4448075/83904536-a3165600-a725-11ea-8c2f-3ec9dcfeaee5.gif)

brad-decker commented on test/helper.js:
```diff
@@ -82,6 +83,21 @@ window.localStorage = {}
 // override metamask-logo
 window.requestAnimationFrame = () => {}
 
+// mock window.matchMedia which is not provided by jsDom
```
Leftover from a previous iteration. i'll remove

brad-decker committed:
display warning inline in fullscreen

The asset list item and transaction list item do not make great use of
screen realestate when in fullscreen mode on larger displays/windows.
This PR adds in a new prop for list-item for displaying content in the
middle of the list-item when on these types of screens. To facilitate
this, some updates were made to styling to allow for the list item to
better control how items are laid out in various breakpoints.

This works makes it possible to display the outdated balance warning
inline in the middle section of the list item, and convert to a tooltip
when on small displays.

brad-decker commented on ui/app/components/app/asset-list-item/asset-list-item.js:
```diff
@@ -15,12 +18,40 @@ const AssetListItem = ({
+    )
+  }, [warning])
+
+  const midContent = useMemo(() => {
```
removed the memoization

Gudahtt replied:
Ah I see, so this is like the React Hooks way of doing what was previously handled by `shouldComponentUpdate`. I recall reading this but I had forgotten, I haven't done that yet. Thanks for explaining!

Gudahtt reviewed (approved):
LGTM!
